python/py-xraylarch/Portfile

```diff
@@ -54,25 +52,34 @@ if {${name} ne ${subport}} {
 
     depends_run-append \
                     port:py${python.version}-asteval \
+                    port:py${python.version}-charset-normalizer \
+                    port:py${python.version}-dill \
                     port:py${python.version}-h5py \
+                    port:py${python.version}-hdf5plugin \
                     port:py${python.version}-fabio \
+                    port:py${python.version}-imageio \
                     port:py${python.version}-importlib-metadata \
                     port:py${python.version}-lmfit \
                     port:py${python.version}-matplotlib \
                     port:py${python.version}-numdifftools \
-                    port:py${python.version}-Pillow \
+                    port:py${python.version}-numpy \
+                    port:py${python.version}-pandas \
                     port:py${python.version}-peakutils \
+                    port:py${python.version}-Pillow \
                     port:py${python.version}-psutil \
+                    port:py${python.version}-pymatgen \
                     port:py${python.version}-pyqt5 \
                     port:py${python.version}-pyqtgraph \
                     port:py${python.version}-pyqt5-webengine \
+                    port:py${python.version}-pyshortcuts \
                     port:py${python.version}-requests \
                     port:py${python.version}-ruamel-yaml-clib \
                     port:py${python.version}-scikit-image \
                     port:py${python.version}-scikit-learn \
                     port:py${python.version}-scipy \
                     port:py${python.version}-silx \
                     port:py${python.version}-sqlalchemy \
```

Comment:
this appears to be missing: [`sqlalchemy_utils`](https://github.com/xraypy/xraylarch/blob/41f1329ed2465faf091e353aa8cd59b66221e78e/setup.cfg#L48C5-L48C21), [`packaging`](https://github.com/xraypy/xraylarch/blob/41f1329ed2465faf091e353aa8cd59b66221e78e/setup.cfg#L54), [`mp_api`](https://github.com/xraypy/xraylarch/blob/41f1329ed2465faf091e353aa8cd59b66221e78e/setup.cfg#L67), [`pycifrw`](https://github.com/xraypy/xraylarch/blob/41f1329ed2465faf091e353aa8cd59b66221e78e/setup.cfg#L68C5-L68C12), [`pyfai`](https://github.com/xraypy/xraylarch/blob/41f1329ed2465faf091e353aa8cd59b66221e78e/setup.cfg#L70C5-L70C10), and [`numexpr`](numexpr).

Please take another look!

Reply:
I can add a few of these (`packaging`, `numexpr`), but there are problems:

- `py-pyFAI` won't build on my Mac. Unsure why. I'll see if the CI system has better luck.
- There are no existing ports for the other three (`sqlalchemy_utils`, `mp_api`, and `pycifrw`).
- In turn, they have at least a few other dependencies that are also not ports (e.g. [mp-api](https://github.com/materialsproject/api/blob/main/pyproject.toml) depends on `maggma` and `emmet-core`).

I do apologize, but I am not willing to run down this entire dependency graph and create 5 (or more!) new ports. This package apparently works just fine when installed by pip, so is there value in having a port? Ideas:

1. We can mark this port as obsolete, letting users just install it from PyPI, or
2. We can keep an incomplete dependency list, knowing that some features will fail because of missing dependencies, or
3. We can try to identify and freeze on an older version that lacks these requirements, or
4. We can hope for a volunteer to make all the required ports.

I see the first as the only workable solution. The second seems unprofessional. The third seems both temporary and weak. The fourth is not a real plan.

Reply:
Update: the [CI system](https://github.com/macports/macports-ports/actions/runs/10217531821/job/28271529277?pr=25157) also fails to build `py-pyFAI`.

Reply:
I think I fixed the problem with `py-pyFAI` (and updated the code by approximately 7.5 years!). See https://github.com/macports/macports-ports/pull/25243/

I'm still not sure about whether I want to go through with adding ports to make xraylarch work. The problem is that the list runs to ten or more, and I'm not done chasing them all down.